src/plugins/http.js

```diff
@@ -138,12 +161,32 @@ function unpatch (http) {
 module.exports = [
   {
     name: 'http',
-    patch,
+    patch: function (http, tracer, config) {
+      patch.call(this, http, 'request', tracer, config)
+      if (semver.satisfies(process.version, '>=8')) {
+        /**
+         * In newer Node versions references internal to modules, such as `http(s).get` calling `http(s).request`, do
+         * not use externally patched versions, which is why we need to also patch `get` here separately.
+         */
+        patch.call(this, http, 'get', tracer, config)
+      }
+    },
     unpatch
   },
   {
     name: 'https',
-    patch,
+    patch: function (http, tracer, config) {
+      if (semver.satisfies(process.version, '>=9')) {
+        patch.call(this, http, 'request', tracer, config)
+        patch.call(this, http, 'get', tracer, config)
+      } else {
+        /**
+         * Below Node v9 the `https` module invokes `http.request`, which would end up counting requests twice.
+         * So rather then patch the `https` module, we ensure the `http` module is patched and we count only there.
+         */
+        require('http')
```

Comment:
If `https` requires `http`, why does it need to be additionally required here?

Reply:
This is probably related to the change described here 
https://github.com/DataDog/dd-trace-js/pull/268/files#r217523774. That is, even though `https` itself loads `http` it’s internally still referring to the unpatched version.

So maybe the Node change is that it makes an HTTP request very early on (before even running the user’s code?), before DataDog has the opportunity to patch anything? 🤔

Reply:
Hmm it should be impossible that it refers to the unpatched version, unless we load `http` at some other point in the tracer before it's instrumented. This happened with other instrumented modules in the past. In that case the module cache should be wiped before the test. I'll take a look to try and find what could be causing this.

Reply:
It seems built-in modules are never stored in the cache, so they cannot be wiped. I'm still not 100% comfortable with this solution, but I don't see any other ways either.

Reply:
I hear you, this type of faffing is why after going down this route the first time I figured to use an approach that did not do conditional patching. But you were right that that made the code a lot more complex.